roadmap/propagate-rust-panic-through-native-frame.md

```diff
@@ -46,3 +48,37 @@ cases (contains destructors, no affordances). We will also have to
 specify this on a per-target basis, as the details will vary depending
 on what exception mechanism is in use.
 
+Another consideration is whether the the native frames include a `catch` block.
+There are several possibilities here as well:
+
+* "catch native" -- e.g. a C++ block catching a specific subset of C++ exceptions
+* "catch all" -- e.g. a C++ catch without an exception argument
+* "catch and release" -- e.g. a C++ "catch all" that re-`throw`s the exception
+
+So far we do **not** have plans to specify these cases.
+
+## Unwind-termination and related concerns
+
+There are several possibilities for the top section of the diagram, above the
+native frames:
+
+* Rust frames - the `panic` re-enters Rust frames
+  * At this point, the `panic` will be no different from a normal `panic`,
+    unless and until it re-enters native frames.
+  * The `panic` may caught as usual, either with `catch_panic` or by crossing a
+    thread boundary.
+* Thread boundary - the exception propagates all the way to a native frame that
+  was invoked from another thread
+* App entry point - the unwind is not caught and does not cross a thread
+  boundary.
+  * This should cause the application to terminate.
+
+Again, these are all currenntly undefined.
```

Comment:
> That clause also makes the "C unwind" ABI no different in practice from the "C" ABI...

No, it makes the ABIs no different in _specification_ (at least temporarily). In practice, we won't actually do this:

> and allows us to emit nounwind for "C unwind" since a program that would actually unwind with that ABI would have UB.

...but we will continue to do so for `"C"`. And, furthermore, we will not leave the `"C unwind"` behavior undefined indefinitely.

Reply:
> No, it makes the ABIs no different in specification (at least temporarily).

Sure, I do not think that an RFC that proposes this can obtain any kind of consensus. Because then this feature introduces real complexity to the spec without adding any value, making it impossible to analyze the tradeoffs. All analysis would need to be hypothetical referring to potential future value that the feature might add. I personally would strongly suggest to defer the RFC until such value can be delivered (note that, implementation wise on nightly, the bar is much lower).